src/parser.rs

```diff
@@ -319,9 +374,47 @@ impl<'a> Parser<'a> {
     fn buffer(&self) -> &[u8] {
         match self.backing {
             Backing::Buf(ref buf) => buf,
-            Backing::Mmap(ref mmap) => unsafe { mmap.as_slice() }
+            Backing::Mmap(ref mmap) => unsafe { mmap.as_slice() },
         }
     }
+
+    fn parse_header(&self) -> Result<Option<Header>> {
+        let mut f = BufReader::new(self.buffer());
+        let mut buf = String::new();
+
+        let mut read = f.read_line(&mut buf)?;
+        if !buf.starts_with('#') {
+            return Ok(None);
+        }
+
+        let mut rv = Header::default();
+        loop {
+            if read == 0 || buf.trim_end().is_empty() {
+                buf.truncate(buf.len() - read);
+                break;
+            }
+            let mut iter = buf.splitn(2, ':');
```

Comment:
i would change the logic to `if !buf.starts_with('#'') { break; }` and then split from `buf[1..]` so that the key can be trimmed below and compared to `compiler` instead of `# compiler`. Pretty sure someone will eventually use different spacing there.